src/components/overlays/badge/Badge.vue

```diff
@@ -0,0 +1,236 @@
+<script setup lang="ts">
+import { type ContextColorsType } from '@/consts/colors';
+import Icon from '@/components/icons/Icon.vue';
+import { type RuiIcons } from '~/src';
+
+export interface Props {
+  text?: string | null;
+  icon?: RuiIcons | null;
+  modelValue?: boolean;
+  dot?: boolean;
+  left?: boolean;
+  offsetX?: string | number;
+  offsetY?: string | number;
+  placement?: 'top' | 'center' | 'bottom';
+  size?: 'sm' | 'md' | 'lg';
+  rounded?: 'sm' | 'md' | 'lg' | 'full';
+  color?: 'default' | ContextColorsType;
+}
+
+defineOptions({
+  name: 'RuiBadge',
+});
+
+const props = withDefaults(defineProps<Props>(), {
+  text: null,
+  icon: null,
+  modelValue: true,
+  dot: false,
+  placement: 'top',
+  size: 'md',
+  rounded: 'full',
+  color: 'primary',
+  offsetX: 0,
+  offsetY: 0,
+});
+
+const css = useCssModule();
+const slots = useSlots();
+
+const { modelValue, offsetX, offsetY, icon, text } = toRefs(props);
+
+const hasIcon = computed(() => !!get(icon) || !!slots.icon);
+
+const hasText = computed(() => !!get(text) || !!slots.badge);
+
+const hasIconAndText = computed(() => get(hasIcon) && get(hasText));
```

Comment:
nitpick: can we use vueuse's `logicAnd` here?

Reply:
Ok